Library/Homebrew/utils/git.rb

```diff
@@ -4,7 +4,7 @@ module Git
   module_function
 
   def last_revision_commit_of_file(repo, file, before_commit: nil)
-    args = [before_commit.nil? ? "--skip=1" : before_commit.split("..").first]
+    args = ["--skip=#{before_commit.nil? ? 1 : before_commit.split("..").first}"]
```

Comment:
That's the correct behaviour. When no commit range is given, it should just skip the latest commit, i.e. `--skip=1`, otherwise, it should pass the first commit of the commit range to `git log`, so it will only show commits before that commit `before_commit`.